### PR TITLE
Fix detect Nuxt3 defineNuxtComponent

### LINKED ISSUE #2057: `vue/order-in-components` is not working with `defineNuxtComponent()`
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.28.0
- **eslint-plugin-vue version:** 9.8.0
- **Node version:** 16.18.1
- **Operating System:** MacOs 12.4 (21F79)

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  extends: ['@antfu', 'plugin:tailwindcss/recommended'],
  plugins: [
    'tailwindcss',
  ],
  rules: {
    'vue/component-tags-order': ['error', {
      order: ['template', 'script', 'style'],
    }],
    'camelcase': ['warn'],
    'import/newline-after-import': ['warn'],
    'no-unused-vars': ['warn'],
    'vue/no-unused-vars': ['warn'],
    'vue/new-line-between-multi-line-property': ['warn', { minLineOfMultilineProperty: 3 }],
    'tailwindcss/classnames-order': 'warn',
    'tailwindcss/no-custom-classname': 0,
    'tailwindcss/no-contradicting-classname': 'error',
    'vue/order-in-components': ['warn', {
      order: [
        'el',
        'name',
        'key',
        'parent',
        'functional',
        ['delimiters', 'comments'],
        ['components', 'directives', 'filters'],
        'extends',
        'mixins',
        ['provide', 'inject'],
        'ROUTER_GUARDS',
        'layout',
        'middleware',
        'validate',
        'scrollToTop',
        'transition',
        'loading',
        'inheritAttrs',
        'model',
        ['props', 'propsData'],
        'emits',
        'setup',
        'asyncData',
        'data',
        'fetch',
        'computed',
        'watch',
        'watchQuery',
        'LIFECYCLE_HOOKS',
        'methods',
        ['template', 'render'],
        'renderError',
        'head',
      ],
    }],
    '@typescript-eslint/space-before-function-paren': ['error', 'always'],
    '@typescript-eslint/brace-style': ['error', '1tbs', { allowSingleLine: true }],
    'vue/quote-props': ['error', 'as-needed'],
    'curly': ['error', 'all'],
    'no-sequences': 0,
    'max-statements-per-line': ['error', { max: 2 }],
  },
}

```

**What did you do?**	

```vue

<script lang="ts">
import VueJsonPretty from 'vue-json-pretty'
import { mapState } from 'pinia'
import { defineNuxtComponent } from '#imports'
import 'vue-json-pretty/lib/styles.css'
import { useOrderStore } from '~/stores/order'
import { useAuthStore } from '~/stores/auth'

export default defineNuxtComponent({
  components: { VueJsonPretty },

  computed: {
    ...mapState(useAuthStore, ['user']),
    ...mapState(useOrderStore, { order: 'data' }),
  },

  data: () => ({
    tab: 'details',
    tabs: {
      details: {
        title: 'Details',
      },

      receipt: {
        title: 'Receipt',
      },
    },
  }),

  head () {
    return {
      title: 'Receipt',
    }
  },
})
</script>

```

**What did you expect to happen?**

It should show warning on line `data: () => ({`

It shows it if I replace `defineNuxtComponent` with `defineComponent`
"ESLint: The "data" property should be above the "computed" property on line 22.(vue/order-in-components)"

**What actually happened?**

No warning.

**Repository to reproduce this issue**

Reproduction:
https://stackblitz.com/edit/github-amdjq7?file=pages/should-warning-but-doesnt.vue
run `yarn lint` in console

### LINKED ISSUE #2191: `vue/no-unused-properties` does not work with `defineNuxtComponent()`
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:**  `8.41.0`
- **eslint-plugin-vue version:** `9.14.0`
- **Node version:** `18`
- **Operating System:** MacOS 13.3.1 (a)（22E772610a）

**Please show your full configuration:**

```json5
module.exports = {
  extends: [
    "plugin:vue/vue3-recommended",
  ],
  rules: {
    "vue/no-unused-properties": [
      "error",
      {
        groups: ["props", "data", "computed", "methods"],
        deepData: false,
        ignorePublicMembers: false,
      },
    ],
  },
};
```

**What did you do?**	

```vue
<template>
  <div>
    <span>Hello</span>
  </div>
</template>

<script lang="ts">
export default defineNuxtComponent({
  name: 'App',
  data() {
    return {
      msg: 'Welcome to Nuxt3', // This data is not used.
    };
  },
});
</script>
```

**What did you expect to happen?**
Data `msg` should be warned like below.
This warning is shown without `defineNuxtComponent`.

スクリーンショット 2023-05-25 21 34 25

It seems to be rerated to https://github.com/vuejs/eslint-plugin-vue/issues/2057.

**What actually happened?**
No warning is shown.

**Repository to reproduce this issue**
https://github.com/uralogical/nuxt3_playground
`feat/add_eslint` is reproduction branch.

### PR DESCRIPTION
**Pull request by sugitata**
* fixes #2191
* fixes #2057

## Description

It's just a workaround like solution.
Add the `defineNuxtComponent` alias for Nuxt3 projects, similar to https://github.com/vuejs/eslint-plugin-vue/pull/1088.

## About defineNuxtComponent

The difference between `defineNuxtComponent` and `defineComponent`  is  only `asyncData` and `head` options are added on `defineNuxtComponent`.
These options are not essential to check in vue-eslint-plugin, so it should be fine to treat it as `defineComponent`.

https://nuxt.com/docs/api/utils/define-nuxt-component

### CODE REVIEW

File: lib/utils/index.js [hunk 1]
```diff
@@ -2558,6 +2558,12 @@ function getVueComponentDefinitionType(node) {
         const isDestructedVueComponent = isObjectArgument(parent)
         return isDestructedVueComponent ? 'defineComponent' : null
       }
+      if (callee.name === 'defineNuxtComponent') {
+        // for Nuxt 3.x
+        // defineNuxtComponent({})
+        const isDestructedVueComponent = isObjectArgument(parent)
+        return isDestructedVueComponent ? 'defineComponent' : null
```
Comment (FloEdelmann) [inline lib/utils/index.js:2565]: Shouldn't this be `defineNuxtComponent`?

```suggestion
        return isDestructedVueComponent ? 'defineNuxtComponent' : null
```

**Review by FloEdelmann — COMMENTED** [review]
Thanks for your contribution!

File: tests/lib/rules/no-undef-components.js [hunk 2]
```diff
@@ -265,6 +265,21 @@ tester.run('no-undef-components', rule, {
         }
       ]
     },
+    {
```
Comment (FloEdelmann) [inline tests/lib/rules/no-undef-components.js:268]: Could you please add a test case for `vue/order-in-components`, too?

**Comment by taksugit** [comment]
@FloEdelmann 
Thank you for your review!
I fixed them.

**Comment by sugitata** [comment]
@FloEdelmann
Thank you for your review!
I fixed them.

**Review by FloEdelmann — APPROVED** [review]
LGTM

**Review by ota-meshi — APPROVED** [review]
LGTM! Sorry for the late review.

**Comment by ota-meshi** [comment]
> Should we consider this a breaking change and keep this for v10?

Yeah. I think this should be included in v10.

**Pull request merged**